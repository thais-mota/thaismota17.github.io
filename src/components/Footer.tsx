import React from 'react';
import { Link } from 'react-router-dom';
import '../styles/footer.scss';

export default function Footer() {
    return (
        <footer className="footer">
            <div>
                <ul>
                    <Link to="/" className="links">
                        <li>Home</li>
                    </Link>
                    <Link to="/about" className="links">
                        <li>About me</li>
                    </Link>
                    <Link to="/projects" className="links">
                        <li>Projects</li>
                    </Link>
                    <Link to="/quotes" className="links">
                        <li>Quotes</li>
                    </Link>
                </ul>
            </div>
            <div>
                <p>Made with &hearts; by Thais Mota</p>
            </div>
        </footer>
    );
}
