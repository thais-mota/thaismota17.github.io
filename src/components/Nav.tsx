import React from 'react';

export default function About() {
    return (
        <nav>
            <p>Navigation bar</p>
        </nav>
    );
}
