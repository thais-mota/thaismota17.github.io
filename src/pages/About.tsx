import React from 'react';

export default function About() {
    return (
        <div>
            <p>Test About Page</p>
        </div>
    );
}
