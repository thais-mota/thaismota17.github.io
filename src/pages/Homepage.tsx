import React from 'react';
import Container from '../components/Container';
import ContainerSeparator, { ContainerSeparatorSize } from '../components/ContainerSeparator';
import List from '../components/List';
import SocialMediaIcons from '../components/SocialMediaIcons';
import Text from '../components/Text';
import Title from '../components/Title';

export default function Homepage() {
    return (
        <Container>
            <>
                <Container>
                    <>
                        <Title>{'Test'}</Title>
                        <ContainerSeparator />
                        <Text>{'Lorem ipsum dolor sit amet, consectetur adipiscing elit' +
                            'sed do eiusmod tempor incididunt ut labore et dolore magna' +
                            'aliqua. Ut enim ad minim veniam, quis nostrud exercitation'}</Text>
                        <ContainerSeparator size={ContainerSeparatorSize.Small} />
                        <SocialMediaIcons />
                        <ContainerSeparator size={ContainerSeparatorSize.Medium} />

                        <Title>{'Things in progress'}</Title>
                        <ContainerSeparator />
                        <List items={['Intern at ZBRA Solutions since March, 2018',
                                      'Bachelors degree in Computer Science at Universidade Anhembi Morumbi since January, 2016 and until December, 2019',
                                      'Learning Piano as a hobby and a passion since 2017',
                                      'Trying to keep a Minimalist Lifestyle',
                                      'Developing this simple portfolio']}/>

                        <Title>{'Future projects'}</Title>
                        <ContainerSeparator />
                        <List items={['Develop a site for an NGO',
                                      'Do the challenge of #100DaysOfCode',
                                      'Improve my english skills',
                                      'Workout daily',
                                      'Read Pragmatic Programmer, by Andrew Hunt and David Thomas',
                                      'Read Effective Java, by Joshua Bloch',
                                      'Read Design Patterns, by Erich Gamma, Richard Helm, Ralph Johnson, John Vlissides',
                                      'Read Introduction to Algorithms, by Thomas H. Cormen and Charles E. Leiserson',
                                      'Read Structure and Interpretation of Computer Programs, by Gerald Jay Sussman and Hal Abelson',
                                      'Read A Vida Imortal de Henrietta Lacks',
                                      'Read Sapiens']}/>

                        <Title>{'Concluded'}</Title>
                        <ContainerSeparator />
                        <List items={['Read Clean Code, by Robert Martin',
                                      'Web Interface course at University of SÃ£o Paulo']}/>
                    </>
                </Container>
            </>
        </Container>
    );
}
