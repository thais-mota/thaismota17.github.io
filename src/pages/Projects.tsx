import React from 'react';

export default function About() {
    return (
        <div>
            <p>Test Projects Page</p>
        </div>
    );
}
