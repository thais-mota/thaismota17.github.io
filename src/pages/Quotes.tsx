import React from 'react';

interface IQuoteProps {
    name: string;
}

export default function Quotes(props: IQuoteProps) {
    return (
        <div>
            <h1>{props.name}</h1>
            <p>Test Quotes Page</p>
        </div>
    );
}
