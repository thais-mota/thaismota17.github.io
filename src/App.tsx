import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.scss';
import Footer from './components/Footer';
import Nav from './components/Nav';
import About from './pages/About';
import Homepage from './pages/Homepage';
import Projects from './pages/Projects';
import Quotes from './pages/Quotes';

export default function App() {
    return (
        <BrowserRouter>
            {/* <Nav /> */}
            <Switch>
                <Route path="/" exact component={Homepage} />
                <Route path="/about" component={About} />
                <Route path="/quotes" component={Quotes} />
                <Route path="/projects" component={Projects} />
            </Switch>
            <Footer />
        </BrowserRouter >
    );
}
