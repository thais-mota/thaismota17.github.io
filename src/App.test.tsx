import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import App from './App';
import Footer from './components/Footer';
import Nav from './components/Nav';
import About from './pages/About';
import Homepage from './pages/Homepage';
import Projects from './pages/Projects';
import Quotes from './pages/Quotes';

test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
});

test('renders Homepage component', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Homepage />
        </MemoryRouter>,
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders About component', () => {
    const component = renderer.create(
        <MemoryRouter>
            <About />
        </MemoryRouter>,
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Quotes component', () => {
    const component = renderer.create(
        <Quotes name="peteca" />,
    );

    expect(component.toJSON()).toMatchSnapshot();
});

test('renders Projects component', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Projects />
        </MemoryRouter>,
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Nav component', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Nav />
        </MemoryRouter>,
    ).toJSON();

    expect(component).toMatchSnapshot();
});

test('renders Footer component', () => {
    const component = renderer.create(
        <MemoryRouter>
            <Footer />
        </MemoryRouter>,
    ).toJSON();

    expect(component).toMatchSnapshot();
});
