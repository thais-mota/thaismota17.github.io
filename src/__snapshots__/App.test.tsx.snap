// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders About component 1`] = `
<div>
  <p>
    Test About Page
  </p>
</div>
`;

exports[`renders Footer component 1`] = `
<footer
  className="footer"
>
  <div>
    <ul>
      <a
        className="links"
        href="/"
        onClick={[Function]}
      >
        <li>
          Home
        </li>
      </a>
      <a
        className="links"
        href="/about"
        onClick={[Function]}
      >
        <li>
          About me
        </li>
      </a>
      <a
        className="links"
        href="/projects"
        onClick={[Function]}
      >
        <li>
          Projects
        </li>
      </a>
      <a
        className="links"
        href="/quotes"
        onClick={[Function]}
      >
        <li>
          Quotes
        </li>
      </a>
    </ul>
  </div>
  <div>
    <p>
      Made with â™¥ by Thais Mota
    </p>
  </div>
</footer>
`;

exports[`renders Homepage component 1`] = `
<nav>
  <p>
    Test Homepage
  </p>
  <p>
    Test Homepage
  </p>
  <p>
    Test Homepage
  </p>
  <p>
    Test Homepage
  </p>
</nav>
`;

exports[`renders Nav component 1`] = `
<nav>
  <p>
    Navigation bar
  </p>
</nav>
`;

exports[`renders Projects component 1`] = `
<div>
  <p>
    Test Projects Page
  </p>
</div>
`;

exports[`renders Quotes component 1`] = `
<div>
  <p>
    Test Quotes Page
  </p>
  <p>
    Test Quotes Page
  </p>
</div>
`;
